src/BSON/Iterator.c

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2014-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <php.h>
+#include <ext/standard/base64.h>
+#include <Zend/zend_interfaces.h>
+#include <Zend/zend_operators.h>
+#include <ext/standard/php_var.h>
+#include <zend_smart_str.h>
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "php_phongo.h"
+#include "phongo_error.h"
+#include "BSON/Iterator.h"
+#include "BSON/Iterator_arginfo.h"
+
+zend_class_entry* php_phongo_iterator_ce;
+
+static bool php_phongo_iterator_init_with_zval(php_phongo_iterator_t* iterator, zval* zbson)
+{
+	php_phongo_document_t*  document_intern;
+	php_phongo_arraylist_t* arraylist_intern;
```

Comment:
Do you actually need these variables? I would think you could just do `bson = Z_DOCUMENT_OBJ_P(&iterator->bson)->bson` below, or at least declare these within the conditional scope.

Reply:
Not really. Removed them.